pkg/services/ngalert/api/tooling/definitions/alertmanager.go

```diff
@@ -1226,6 +1227,9 @@ func (m *ObjectMatchers) UnmarshalYAML(unmarshal func(interface{}) error) error
 			return fmt.Errorf("unsupported match type %q in matcher", rawMatcher[1])
 		}
 
+		rawMatcher[2] = strings.TrimPrefix(rawMatcher[2], "\"")
```

Comment:
Don't be fooled by the simplicity of this solution - I actually tried several approaches before I landed here. As an example I tried:

- [Copying & modifying the Alertmanager's](https://github.com/prometheus/alertmanager/blob/main/pkg/labels/parse.go#L55-L178) `ParseMatchers` and `ParseMatcher` 
- [Borrwing the value unescaping from the `ParseMatcher`](https://github.com/prometheus/alertmanager/blob/main/pkg/labels/parse.go#L139-L171)
- Doing clever things and unmarshalling and marshalling

In the end, all of them either required more work or the solution ended up looking more complex. Instead, I'm going to focus my efforts on trying to separate the logic of `ParseMatchers` and `ParseMatcher` upstream so that we can re-use them upstream. I filed, to keep track of: https://github.com/grafana/grafana/issues/50040